README.md

```diff
@@ -39,66 +60,154 @@ These components collectively make up the SDK, and you should make any necessary
+
+```shell
+KINDE_DOMAIN=https://<replace>.kinde.com
+KINDE_CLIENT_ID=<replace>
+KINDE_CLIENT_SECRET=<replace>
+KINDE_REDIRECT_URI=http://localhost:8080/kinde-j2ee-app/login
 ```
-mvn install:install-file -Dfile=/Users/rahoolkumar/Downloads/international-kinde-java/target/kindejava.jar -DgroupId=org.openapitools -DartifactId=openapi-spring -Dversion=1 -Dpackaging=jar
+
+#### Server Example
+
+To make an M2M server token request onto Kinde, first set up the appropriate environmental variables
+
+##### By shell export
+
+Run these exports before running your service.
+
+```shell
+export KINDE_DOMAIN=https://<replace>.kinde.com # This is the domain you setup at kinde
+export KINDE_CLIENT_ID=<replace> # the id for the client connecting to Kinde
+export KINDE_CLIENT_SECRET=<replace> # the secret used to authenticate the client against Kinde
 ```
-Make sure to replace ``` /Users/rahoolkumar/Downloads/international-kinde-java/target/kindejava.jar``` with the actual path to your JAR file.
 
-In the pom.xml of the other project where you want to use this dependency, add the following dependency entry:
+##### By .env file config
+
+Place this .env file in the directory from which you run your service.
+
+```shell
+KINDE_DOMAIN=https://<replace>.kinde.com
+KINDE_CLIENT_ID=<replace>
+KINDE_CLIENT_SECRET=<replace>
 ```
-<dependency>
-    <groupId>org.openapitools</groupId>
-    <artifactId>openapi-spring</artifactId>
-    <version>1</version>
-    <scope>compile</scope>
-</dependency>
+##### Programmatic configuration
+
+If you want to pass in configuration programmatically the KindeClientBuilder supports this use the following approach.
+```java
+KindeClient kindeClient = KindeClientBuilder
+        .builder()
+        .domain("<replace>")
+        .clientId("<replace>")
+        .clientSecret("<replace>")
+        .build();
 ```
-Save the pom.xml file.
 
-Maven will automatically download the dependency from your local repository and make it available in your project.
+##### Java Code to retrieve a M2M token.
+The example below details how to implement a server level token request. This is needed for M2M communication and authorization.
+```java
+KindeClient kindeClient = KindeClientBuilder
+        .builder()
+        .build();
+KindeClientSession kindeClientSession = kindeClient.clientSession();
+List<KindeToken> tokens = kindeClientSession.retrieveTokens();
+        
+```
+
+### User Code Authorization Example (PKCE)
+
+To authenticate a user on a client, the appropriate configuration must be in place.
+
+##### By shell export
+
+Run these exports before running your service.
+
+```shell
+export KINDE_DOMAIN=https://<replace>.kinde.com # This is the domain you setup at kinde
+export KINDE_CLIENT_ID=<replace> # the id for the client connecting to Kinde
+export KINDE_CLIENT_SECRET=<replace> # the secret used to authenticate the client against Kinde
+export KINDE_REDIRECT_URI=openid # the open id
```

Comment:
This should have < replace >. It's not a redirect uri.

Reply:
Update

Reply:
Updated

Reply:
Done